discard.cc

```diff
@@ -91,17 +91,19 @@ void DiscardFilter::fix_process_wpt(const Waypoint* wpt)
     switch (what) {
     case wptdata:
       waypt_del(waypointp);
+      delete waypointp;
```

Comment:
Offhand, what do you think of releasing the object in what was "obviously" meant to be the destructors instead of leaving it live and being released in the callers? It seems like a hazard that these are still dangling and live after calling something named "delete"